includes/class-wc-payments.php

```diff
@@ -965,6 +971,19 @@ function ( $container ) {
 					}
 				);
 				$blocks_package_container->get( Automattic\WooCommerce\Blocks\Payments\Api::class );
+
+				// Register draft orders.
+				$draft_orders = $blocks_package_container->get( Automattic\WooCommerce\Blocks\Domain\Services\DraftOrders::class );
+
+				add_filter( 'wc_order_statuses', [ $draft_orders, 'register_draft_order_status' ] );
+				add_filter( 'woocommerce_register_shop_order_post_statuses', [ $draft_orders, 'register_draft_order_post_status' ] );
+				add_filter( 'woocommerce_analytics_excluded_order_statuses', [ $draft_orders, 'append_draft_order_post_status' ] );
+				add_filter( 'woocommerce_valid_order_statuses_for_payment', [ $draft_orders, 'append_draft_order_post_status' ] );
+				add_filter( 'woocommerce_valid_order_statuses_for_payment_complete', [ $draft_orders, 'append_draft_order_post_status' ] );
+				// Hook into the query to retrieve My Account orders so draft status is excluded.
+				add_action( 'woocommerce_my_account_my_orders_query', [ $draft_orders, 'delete_draft_order_post_status_from_args' ] );
+				add_action( 'woocommerce_cleanup_draft_orders', [ $draft_orders, 'delete_expired_draft_orders' ] );
+				add_action( 'admin_init', [ $draft_orders, 'install' ] );
```

Comment:
I think you should be set by just doing `$blocks_package_container->get( Automattic\WooCommerce\Blocks\Domain\Services\DraftOrders::class )->init();`
```suggestion
				// Register draft orders.
				$blocks_package_container->get( Automattic\WooCommerce\Blocks\Domain\Services\DraftOrders::class )->init();
```

Reply:
@senadir It didn't work 🙁. `DraftOrders`'s `init` still has [that check](https://github.com/woocommerce/woocommerce-blocks/pull/6288/files#diff-68c030b619e2ef76a61a2efcfa05591c73406fc916889d9a2a2756f461b40338L40) on older versions.

Reply:
Ah, I forget it's the init behind the check, not the actual class.